c3: update get started guides with info on deploying with git repos. (#12966)
cloudflare/cloudflare-docs: content/pages/get-started/c3.md

To get started, open a terminal window and run:

{{<tabs labels="npm | yarn | pnpm | bun">}}
{{<tab label="npm" default="true">}}
```sh
$ npm create cloudflare@latest
```
{{</tab>}}
{{<tab label="yarn">}}
```sh
$ yarn create cloudflare
```
{{</tab>}}
{{<tab label="pnpm">}}
```sh
$ pnpm create cloudflare@latest
```
{{</tab>}}
{{<tab label="bun">}}
```sh
$ bun create cloudflare@latest
```
{{</tab>}}
{{</tabs>}}

> Running this command will prompt you to install the [`create-cloudflare`](https://www.npmjs.com/package/create-cloudflare) package, and lead you through a setup wizard. After you have entered the setup wizard, you will be asked which type of application you would like to create.

Reviewer: ```suggestion
Running `create cloudflare@latest` will prompt you to install the [`create-cloudflare`](https://www.npmjs.com/package/create-cloudflare) package, and lead you through a setup wizard. After you have entered the setup wizard, you will be asked which type of application you would like to create.
```

This is the full format of a C3 invocation alongside the possible CLI arguments:

> {{<tabs labels="npx | yarn | pnpm | bun">}}

Reviewer: This should say `npm`, not `npx`.`npx` is a way to run packages installed with `npm` and so, `npx` is not a standalone package manager like the rest of the options in the tab switcher, it's still under `npm` umbrella.

{{<tab label="npx" default="true">}}
```sh

> $ npx create-cloudflare@latest [--] [<DIRECTORY>] [OPTIONS] [-- <NESTED ARGS...>]

Reviewer: 1. Could we double check if it is `npm` or `npx` in front of `create cloudflare@latest`? I believe the right command is`npm` because it's prompting an install but we should be consistent here. In Workers get started, it is `npm`.

2. I think there is an unnecessary dash `-` between `create` and `cloudflare` here as well.

Reviewer: @penalosa just told me both are technically work, but can we standardize `npm create cloudflare@ latest` to unify with what is in the Workers docs?

Author: I've updated all examples to use `npm`.

    - `webFramework`: A website or web application.
    - `hello-world`: A basic "Hello World" Cloudflare Worker.

>     - `hello-world-durable-object`: A basic "Hello World" Cloudflare Worker.

Reviewer: This seems incorrect.

    - `common`: A Cloudflare Worker which implements a common example of routing/proxying functionalities.
    - `scheduled`: A scheduled Cloudflare Worker (triggered via [Cron Triggers](/workers/configuration/cron-triggers/)).
    - `queues`: A Cloudflare Worker which is both a consumer and produced of [Queues](/queues/).
    - `chatgptPlugin`: A ChatGPT plugin.
    - `openapi`: A Worker implementing an OpenAPI REST endpoint.
    - `remote-template`: Create a new project from a git repository.

>     - `pre-existing`: Fetch a Worker initialized from the dashboard.

Reviewer: ```suggestion
    - `pre-existing`: Fetch a Worker initialized from the Cloudflare dashboard.
```

    - `svelte`
    - `vue`

- `--template` {{<type>}}string{{</type>}}

>   - A git repository used to create a new project. At a minimum, template folders must contain the following:

Reviewer: ```suggestion
  - A Git repository used to create a new project. At a minimum, template folders must contain the following:
```

    - `package.json`
    - `wrangler.toml`

>     - `src/` containing a worker script referenced from `wrangler.toml`

Reviewer: ```suggestion
    - `src/` containing a Worker referenced from `wrangler.toml`.
```
